### PR TITLE
Move the helmer structure to the api folder

### PR DESCRIPTION
**Pull request by SchSeba**
This is needed for other projects that want to use the api.

Before they pull the helmer package and not only the api package
and before we are adding patches to the helm package inside vendor
the application was not able to compile.

error example:
```
go vet ./cnf-tests/testsuites/...
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:292:20: install.ReplaceRelease undefined (type *action.Install has no field or method ReplaceRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:311:21: install.FailRelease undefined (type *action.Install has no field or method FailRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:328:20: install.FailRelease undefined (type *action.Install has no field or method FailRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:336:21: install.FailRelease undefined (type *action.Install has no field or method FailRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:347:19: install.RecordRelease undefined (type *action.Install has no field or method RecordRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:409:25: ActionConfig.DeleteHookByPolicy undefined (type *action.Configuration has no field or method DeleteHookByPolicy)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:417:15: ActionConfig.RecordRelease undefined (type *action.Configuration has no field or method RecordRelease)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:428:26: ActionConfig.DeleteHookByPolicy undefined (type *action.Configuration has no field or method DeleteHookByPolicy)
vendor/github.com/openshift-psap/special-resource-operator/pkg/helmer/helmer.go:443:25: ActionConfig.DeleteHookByPolicy undefined (type *action.Configuration has no field or method DeleteHookByPolicy)

```

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Comment by zvonkok** [comment]
Before running go vet, can you run a `make patch`  and then run `go vet` ?

**Comment by SchSeba** [comment]
@zvonkok tested

```
make patch               
which: no kube-linter in (/home/sscheink/.cargo/bin:/usr/lib64/qt-3.3/bin:/home/sscheink/.cargo/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:/usr/local/go/bin:/home/sscheink/Documents/GolangProjects/bin/:/usr/local/kubebuilder/bin:/home/sscheink/.local/bin:/home/sscheink/bin:/home/sscheink/Documents/GolangProjects/bin/:/usr/local/kubebuilder/bin:/home/sscheink/.cargo/bin)
cp .patches/options.patch.go vendor/github.com/google/go-containerregistry/pkg/crane/.
cp .patches/getter.patch.go vendor/helm.sh/helm/v3/pkg/getter/.
cp .patches/action.patch.go vendor/helm.sh/helm/v3/pkg/action/.
cp .patches/install.patch.go vendor/helm.sh/helm/v3/pkg/action/.
go vet ./...
<clean>
```

**Comment by SchSeba** [comment]
Hi @zvonkok the failing lane looks like a flake no?

File: pkg/helmer/api/v1beta1/helmer_types.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	*out = *in
+	out.Repository = in.Repository
+	if in.Tags != nil {
+		in, out := &in.Tags, &out.Tags
+		*out = make([]string, len(*in))
+		copy(*out, *in)
+	}
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new HelmChart.
```
Comment (dagrayvid) [inline pkg/helmer/api/v1beta1/helmer_types.go]:
```suggestion
// DeepCopy is a manually created deepcopy function, copying the receiver, creating a new HelmChart.
```

File: pkg/helmer/api/v1beta1/helmer_types.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+func (in *HelmChart) DeepCopy() *HelmChart {
+	if in == nil {
+		return nil
+	}
+	out := new(HelmChart)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be nonnil.
```
Comment (dagrayvid) [inline pkg/helmer/api/v1beta1/helmer_types.go]:
```suggestion
// DeepCopyInto is a manually created deepcopy function, copying the receiver, writing into out. in must be nonnil.
```

File: pkg/helmer/api/v1beta1/helmer_types.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+func (in *HelmRepo) DeepCopyInto(out *HelmRepo) {
+	*out = *in
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new HelmRepo.
```
Comment (dagrayvid) [inline pkg/helmer/api/v1beta1/helmer_types.go]:
```suggestion
// DeepCopy is a manually created deepcopy function, copying the receiver, creating a new HelmRepo.
```

**Pull request merged**